specification.markdown

```diff
@@ -386,41 +386,60 @@ The path defines the allowed storage locations of the EBML Element within an EBM
-EBMLParentPath        = EBMLFixedParent EBMLLastParent
-EBMLFixedParent       = *(EBMLPathAtom)
-EBMLElementPath       = EBMLPathAtom / EBMLPathAtomRecursive
-EBMLPathAtom          = PathDelimiter EBMLAtomName
-EBMLPathAtomRecursive = "(1*(" EBMLPathAtom "))"
-EBMLLastParent        = EBMLPathAtom / EBMLGlobalParent
-EBMLGlobalParent      = "(" GlobalParentOccurence "\)"
-EBMLAtomName          = 1*(EBMLNameChar)
-EBMLNameChar          = ALPHA / DIGIT / "-" / "."
+EBMLFullPath          = EBMLParentPath EBMLElement
+
+EBMLParentPath        = PathDelimiter [EBMLParents]
+
+EBMLParents           = 0*IntermediatePathAtom EBMLLastParent
+IntermediatePathAtom  = EBMLPathAtom / GlobalPlaceholder
+EBMLLastParent        = EBMLPathAtom / GlobalPlaceholder
+
+EBMLPathAtom          = [IsRecursive] EBMLAtomName PathDelimiter
+EBMLElement           = [IsRecursive] EBMLAtomName
+
 PathDelimiter         = "\"
-GlobalParentOccurence = [PathMinOccurrence] "*" [PathMaxOccurrence]
+IsRecursive           = "+"
+EBMLAtomName          = ALPHA / DIGIT 0*EBMLNameChar
+EBMLNameChar          = ALPHA / DIGIT / "-" / "."
+
+GlobalPlaceholder     = "(" GlobalParentOccurence "\)"
+GlobalParentOccurence = [PathMinOccurrence] "-" [PathMaxOccurrence]
 PathMinOccurrence     = 1*DIGIT ; no upper limit
 PathMaxOccurrence     = 1*DIGIT ; no upper limit
 ```
 
 The `*`, `(` and `)` symbols are interpreted as defined in [@!RFC5234].
 
-The EBMLPathAtom part of the EBMLElementPath MUST be equal to the name attribute of the EBML Schema.
+The EBMLAtomName of the EBMLElement part MUST be equal to the `@name` attribute of the EBML Schema.
+If the EBMLElement part contains an IsRecursive part, the EBML Element can occur within itself recursively (see (#recursive)).
 
-The starting PathDelimiter of the path corresponds to the root of the EBML Document.
+The starting PathDelimiter of EBMLParentPath corresponds to the root of the EBML Document.
 
-In some cases the EBMLLastParent part of the path is an EBMLGlobalParent. A path with a EBMLGlobalParent defines a Global Element; see (#global-elements). Any path that starts with the EBMLFixedParent of the Global Element and matches the occurrences found in the GlobalParentOccurence is a valid path for the Global Element. 
+The `@path` value MUST be unique within the EBML Schema. The `@id` value corresponding to this `@path` MUST NOT be defined for use within another EBML Element with the same EBMLParentPath as this `@path`.
 
-The GlobalParentOccurence part is interpreted as an ABNF Variable Repetition. The repetition amounts correspond to the amount of unspecified Parent Element levels there can be between the EBMLFixedParent and the actual EBMLElementPath.
+A path with a GlobalPlaceholder as the EBMLLastParent defines a Global Element; see (#global-elements).
+If the element has no EBMLLastParent part or the EBMLLastParent part is not a GlobalPlaceholder then the Element is not a Global Element.
```

Comment:
Maybe more logical to say `If the element has no EBMLParents part or...`, but still confused.

Suppose you have a Global Master element like:

`/(0-1)/GlobalMaster`

then a Child Element of that would be:
`/(0-1)/GlobalMaster/SubElement`

With this definition, `SubElement` is not considered as a Global Element. Is that intended?

Reply:
That's correct. The SubElement doesn't get to be placed within all kinds of various unknown parents. It has only a fixed parent.